name-service/js/test/unit/index.test.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  Keypair,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  SystemProgram,
+} from '@solana/web3.js';
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import {
+  createInstruction,
+  deleteInstruction,
+  reallocInstruction,
+  transferInstruction,
+  updateInstruction,
+} from '../../src';
+import { Numberu32, Numberu64 } from '../../src/utils';
+
+chai.use(chaiAsPromised);
+
+describe('SplNameService Instructions', () => {
+  const nameServiceAddress = new PublicKey(
+    'namesLPneVptA9Z5rqUDD9tMTWEJwofgaYwp8cawRkX'
+  );
+  const nameAccountKey = Keypair.generate().publicKey;
+  const nameOwnerKey = Keypair.generate().publicKey;
+  const payerKey = Keypair.generate().publicKey;
+  const name = Buffer.from('hello');
+
+  it('createInstruction', () => {
+    const instruction = createInstruction(
+      nameServiceAddress,
+      SystemProgram.programId,
+      nameAccountKey,
+      nameOwnerKey,
+      payerKey,
+      name,
+      new Numberu64(LAMPORTS_PER_SOL),
+      new Numberu64(10)
+    );
+
+    expect(instruction.keys).to.have.length(6);
+    instruction.keys[0].pubkey.equals(SystemProgram.programId);
+    instruction.keys[1].pubkey.equals(payerKey);
+    instruction.keys[2].pubkey.equals(nameAccountKey);
+    instruction.keys[3].pubkey.equals(nameOwnerKey);
+    instruction.keys[4].pubkey.equals(new PublicKey(Buffer.alloc(32)));
+    instruction.keys[5].pubkey.equals(new PublicKey(Buffer.alloc(32)));
```

Comment:
For completeness, can you add a test that uses nameClassKey / nameParent / nameParentOwner and checks that they're properly filled in here?